File: src/delaunay/quad_edge.rs
pub(crate) type QuadEdge = usize;

pub(super) fn to_opposite_edge(edge: QuadEdge) -> QuadEdge {
    ((edge >> 2) << 2) + ((edge + 2) & 3)
}

pub(super) fn to_rotated_edge(edge: QuadEdge) -> QuadEdge {
    ((edge >> 2) << 2) + ((edge + 1) & 3)
}

File: rene/_delaunay/quad_edge.py
import sys
from typing import NewType

QuadEdge = NewType('QuadEdge', int)

UNDEFINED_EDGE = QuadEdge(sys.maxsize)


def to_opposite_edge(edge: QuadEdge) -> QuadEdge:
    """
    aka "Sym" in L. Guibas and J. Stolfi notation.
    """
    return QuadEdge(((edge >> 2) << 2) + ((edge + 2) & 3))


def to_rotated_edge(edge: QuadEdge) -> QuadEdge:
    """
    aka "Rot" in L. Guibas and J. Stolfi notation.
    """
    return QuadEdge(((edge >> 2) << 2) + ((edge + 1) & 3))
